src/libraries/System.IO.FileSystem/tests/Directory/SetCurrentDirectory.cs

```diff
@@ -33,10 +33,15 @@ public void SetToValidOtherDirectory()
             RemoteExecutor.Invoke(() =>
             {
                 Directory.SetCurrentDirectory(TestDirectory);
-                // On OSX, the temp directory /tmp/ is a symlink to /private/tmp, so setting the current
-                // directory to a symlinked path will result in GetCurrentDirectory returning the absolute
-                // path that followed the symlink.
-                if (!OperatingSystem.IsMacOS())
+
+                if ((File.GetAttributes(Path.GetDirectoryName(TestDirectory)) & FileAttributes.ReparsePoint) != 0)
```

Comment:
This file has the comprehensive current directory tests, so it gets more complicated checks unlike the other places.